app/qml/components/MMSwitch.qml

```diff
@@ -16,9 +16,19 @@ Switch {
 
   property string textOn
   property string textOff
+  property bool visibleText: true
 
   contentItem: Text {
-    text: (control.textOn.length > 0 && control.textOff.length > 0) ? (control.checked ? control.textOn : control.textOff) : control.text
+    text: {
+      if(control.visibleText) {
+        if(control.textOn.length > 0 && control.textOff.length > 0) {
+          if(control.checked)
+            return control.textOn
+          return control.textOff
+        }
+        return control.text
+      }
+    }
```

Comment:
It will be better to not bother with `textOn` and `textOff`, but have only `text`. Whoever uses the component will bind the `text` property themself